### ISSUE TITLE
[2.x] test failures in Render suite: Assert 'Check SDL_CreateRenderer result': Failed

### ISSUE THREAD

**Issue by smcv** [libsdl-org/SDL#8799]
## Steps to reproduce

The same preconditions as in #8796:

```
$ git describe --tags
release-2.28.0-362-g5e12e4b81
$ podman run --rm -it -w $(pwd) -v $(pwd):$(pwd):rw -v $(pwd)/.git:$(pwd)/.git:ro debian:sid-slim
# sed -i -e 's/Types: deb/& deb-src/' /etc/apt/sources.list.d/debian.sources
# apt update
# apt upgrade
# apt build-dep libsdl2
# ./configure
# make
# cd test
# ./configure \
  --x-includes=/usr/include \
  --x-libraries=/usr/lib/x86_64-linux-gnu \
  SDL_CFLAGS=-I$(pwd)/../include \
  SDL_LIBS="-L$(pwd)/../build/.libs -lSDL2" \
  ac_cv_lib_SDL2_ttf_TTF_Init=no
# make
# export SDL_VIDEODRIVER=dummy SDL_AUDIODRIVER=dummy LD_LIBRARY_PATH=$(pwd)/../build/.libs
# make check
```

(the steps above are similar to how Debian currently builds SDL2)

Still in the same container as above, run

```
# ./testautomation
# ./testautomation --filter Render
```

## Expected result

All tests should ideally pass, both `make check` and `testautomation`.

## Actual result

`make check` passes.

`testautomation` fails with some errors reported elsewhere (#8796, #8797). ~~but the whole Render suite passes.~~ Tests in the Render suite superficially appear to succeed (no instructions for re-running failing render tests are provided in the summary at the end), but actually, the Render suite's setup function is failing.

It's easier to see this if you re-run with `--filter Render`:

```
INFO:  01/08/24 17:02:09: ::::: Test Run /w seed 'AH70KXLAIAB6OK8V' started
INFO:  01/08/24 17:02:09: Filtering: running only suite 'Render'
INFO:  01/08/24 17:02:09: ===== Test Suite 1: 'Audio' skipped
INFO:  01/08/24 17:02:09: ===== Test Suite 2: 'Clipboard' skipped
INFO:  01/08/24 17:02:09: ===== Test Suite 3: 'Events' skipped
INFO:  01/08/24 17:02:09: ===== Test Suite 4: 'GUID' skipped
INFO:  01/08/24 17:02:09: ===== Test Suite 5: 'Hints' skipped
INFO:  01/08/24 17:02:09: ===== Test Suite 6: 'Joystick' skipped
INFO:  01/08/24 17:02:09: ===== Test Suite 7: 'Keyboard' skipped
INFO:  01/08/24 17:02:09: ===== Test Suite 8: 'Main' skipped
INFO:  01/08/24 17:02:09: ===== Test Suite 9: 'Math' skipped
INFO:  01/08/24 17:02:09: ===== Test Suite 10: 'Mouse' skipped
INFO:  01/08/24 17:02:09: ===== Test Suite 11: 'Pixels' skipped
INFO:  01/08/24 17:02:09: ===== Test Suite 12: 'Platform' skipped
INFO:  01/08/24 17:02:09: ===== Test Suite 13: 'Rect' skipped
INFO:  01/08/24 17:02:09: ===== Test Suite 14: 'Render' started
INFO:  01/08/24 17:02:09: ----- Test Case 14.1: 'render_testGetNumRenderDrivers' started
INFO:  01/08/24 17:02:09: Test Description: 'Tests call to SDL_GetNumRenderDrivers'
INFO:  01/08/24 17:02:09: Test Iteration 1: execKey 9925300554255249160
INFO:  01/08/24 17:02:09: Assert 'SDL_CreateWindow()': Passed
INFO:  01/08/24 17:02:09: Assert 'Check SDL_CreateWindow result': Passed
INFO:  01/08/24 17:02:09: Assert 'SDL_CreateRenderer()': Passed
ERROR: 01/08/24 17:02:09: Assert 'Check SDL_CreateRenderer result': Failed
ERROR: 01/08/24 17:02:09: >>> Suite Setup 'Render': Failed
INFO:  01/08/24 17:02:09: Total Test runtime: 0.0 sec
INFO:  01/08/24 17:02:09: ----- Test Case 14.2: 'render_testPrimitives' started
INFO:  01/08/24 17:02:09: Test Description: 'Tests rendering primitives'
INFO:  01/08/24 17:02:09: Test Iteration 1: execKey 10880306080719626408
INFO:  01/08/24 17:02:09: Assert 'SDL_CreateWindow()': Passed
INFO:  01/08/24 17:02:09: Assert 'Check SDL_CreateWindow result': Passed
INFO:  01/08/24 17:02:09: Assert 'SDL_CreateRenderer()': Passed
ERROR: 01/08/24 17:02:09: Assert 'Check SDL_CreateRenderer result': Failed
ERROR: 01/08/24 17:02:09: >>> Suite Setup 'Render': Failed
INFO:  01/08/24 17:02:09: Total Test runtime: 0.0 sec
INFO:  01/08/24 17:02:09: ----- Test Case 14.3: 'render_testPrimitivesBlend' started
INFO:  01/08/24 17:02:09: Test Description: 'Tests rendering primitives with blending'
INFO:  01/08/24 17:02:09: Test Iteration 1: execKey 1566760837620923259
INFO:  01/08/24 17:02:09: >>> Test 'render_testPrimitivesBlend': Skipped (Disabled)
INFO:  01/08/24 17:02:09: Total Test runtime: 0.0 sec
INFO:  01/08/24 17:02:09: ----- Test Case 14.4: 'render_testBlit' started
INFO:  01/08/24 17:02:09: Test Description: 'Tests blitting'
INFO:  01/08/24 17:02:09: Test Iteration 1: execKey 1997885719956063826
INFO:  01/08/24 17:02:09: Assert 'SDL_CreateWindow()': Passed
INFO:  01/08/24 17:02:09: Assert 'Check SDL_CreateWindow result': Passed
INFO:  01/08/24 17:02:09: Assert 'SDL_CreateRenderer()': Passed
ERROR: 01/08/24 17:02:09: Assert 'Check SDL_CreateRenderer result': Failed
ERROR: 01/08/24 17:02:09: >>> Suite Setup 'Render': Failed
INFO:  01/08/24 17:02:09: Total Test runtime: 0.0 sec
INFO:  01/08/24 17:02:09: ----- Test Case 14.5: 'render_testBlitColor' started
INFO:  01/08/24 17:02:09: Test Description: 'Tests blitting with color'
INFO:  01/08/24 17:02:09: Test Iteration 1: execKey 3062308137428117473
INFO:  01/08/24 17:02:09: Assert 'SDL_CreateWindow()': Passed
INFO:  01/08/24 17:02:09: Assert 'Check SDL_CreateWindow result': Passed
INFO:  01/08/24 17:02:09: Assert 'SDL_CreateRenderer()': Passed
ERROR: 01/08/24 17:02:09: Assert 'Check SDL_CreateRenderer result': Failed
ERROR: 01/08/24 17:02:09: >>> Suite Setup 'Render': Failed
INFO:  01/08/24 17:02:09: Total Test runtime: 0.0 sec
INFO:  01/08/24 17:02:09: ----- Test Case 14.6: 'render_testBlitAlpha' started
INFO:  01/08/24 17:02:09: Test Description: 'Tests blitting with alpha'
INFO:  01/08/24 17:02:09: Test Iteration 1: execKey 13554164796235776514
INFO:  01/08/24 17:02:09: >>> Test 'render_testBlitAlpha': Skipped (Disabled)
INFO:  01/08/24 17:02:09: Total Test runtime: 0.0 sec
INFO:  01/08/24 17:02:09: ----- Test Case 14.7: 'render_testBlitBlend' started
INFO:  01/08/24 17:02:09: Test Description: 'Tests blitting with blending'
INFO:  01/08/24 17:02:09: Test Iteration 1: execKey 6925310478108695064
INFO:  01/08/24 17:02:09: >>> Test 'render_testBlitBlend': Skipped (Disabled)
INFO:  01/08/24 17:02:09: Total Test runtime: 0.0 sec
INFO:  01/08/24 17:02:09: Total Suite runtime: 0.0 sec
ERROR: 01/08/24 17:02:09: Suite Summary: Total=7 Passed=0 Failed=4 Skipped=3
ERROR: 01/08/24 17:02:09: >>> Suite 'Render': Failed
INFO:  01/08/24 17:02:09: ===== Test Suite 15: 'RWops' skipped
INFO:  01/08/24 17:02:09: ===== Test Suite 16: 'SDLtest' skipped
INFO:  01/08/24 17:02:09: ===== Test Suite 17: 'Stdlib' skipped
INFO:  01/08/24 17:02:09: ===== Test Suite 18: 'Surface' skipped
INFO:  01/08/24 17:02:09: ===== Test Suite 19: 'SysWM' skipped
INFO:  01/08/24 17:02:09: ===== Test Suite 20: 'Timer' skipped
INFO:  01/08/24 17:02:09: ===== Test Suite 21: 'Video' skipped
INFO:  01/08/24 17:02:09: ===== Test Suite 22: 'Subsystems' skipped
INFO:  01/08/24 17:02:09: Total Run runtime: 0.0 sec
ERROR: 01/08/24 17:02:09: Run Summary: Total=7 Passed=0 Failed=4 Skipped=3
ERROR: 01/08/24 17:02:09: >>> Run /w seed 'AH70KXLAIAB6OK8V': Failed
INFO:  01/08/24 17:02:09: Exit code: 1
```

~~Is this perhaps relying on some initialization having "leaked" in from an earlier test-case? Or do I need to be selecting a specific dummy renderer to go with my `SDL_VIDEODRIVER=dummy`?~~

**Comment by slouken** [comment]
Is running the render tests standalone something you need for Debian acceptance testing? Does it work with sdl2-compat?

**Comment by slouken** [comment]
Yes, that makes sense. I'll make sure the test suite as a whole runs with dummy audio and video for 2.30 release. I'm not going to worry about individual suite failures for now. sdl2-compat should pass at least as much as the real SDL2.

**Comment by smcv** [comment]
Hmm, this might not be limited to just running with `--filter Render`. If I patch the test suite to change the three tests from #8797 and one test from #8796 to `TEST_DISABLED`, and run `./testautomation` with no filter, then I still get

```
INFO:  01/08/24 18:14:01: ===== Test Suite 14: 'Render' started
INFO:  01/08/24 18:14:01: ----- Test Case 14.1: 'render_testGetNumRenderDrivers' started
INFO:  01/08/24 18:14:01: Test Description: 'Tests call to SDL_GetNumRenderDrivers'
INFO:  01/08/24 18:14:01: Test Iteration 1: execKey 4578152224225714046
INFO:  01/08/24 18:14:01: Assert 'SDL_CreateWindow()': Passed
INFO:  01/08/24 18:14:01: Assert 'Check SDL_CreateWindow result': Passed
INFO:  01/08/24 18:14:01: Assert 'SDL_CreateRenderer()': Passed
ERROR: 01/08/24 18:14:01: Assert 'Check SDL_CreateRenderer result': Failed
ERROR: 01/08/24 18:14:01: >>> Suite Setup 'Render': Failed
INFO:  01/08/24 18:14:01: Total Test runtime: 0.0 sec
INFO:  01/08/24 18:14:01: ----- Test Case 14.2: 'render_testPrimitives' started
INFO:  01/08/24 18:14:01: Test Description: 'Tests rendering primitives'
INFO:  01/08/24 18:14:01: Test Iteration 1: execKey 2873045829393285211
INFO:  01/08/24 18:14:01: Assert 'SDL_CreateWindow()': Passed
INFO:  01/08/24 18:14:01: Assert 'Check SDL_CreateWindow result': Passed
INFO:  01/08/24 18:14:01: Assert 'SDL_CreateRenderer()': Passed
ERROR: 01/08/24 18:14:01: Assert 'Check SDL_CreateRenderer result': Failed
ERROR: 01/08/24 18:14:01: >>> Suite Setup 'Render': Failed
INFO:  01/08/24 18:14:01: Total Test runtime: 0.0 sec
INFO:  01/08/24 18:14:01: ----- Test Case 14.3: 'render_testPrimitivesBlend' started
INFO:  01/08/24 18:14:01: Test Description: 'Tests rendering primitives with blending'
INFO:  01/08/24 18:14:01: Test Iteration 1: execKey 5960817489374515484
INFO:  01/08/24 18:14:01: >>> Test 'render_testPrimitivesBlend': Skipped (Disabled)
INFO:  01/08/24 18:14:01: Total Test runtime: 0.0 sec
INFO:  01/08/24 18:14:01: ----- Test Case 14.4: 'render_testBlit' started
INFO:  01/08/24 18:14:01: Test Description: 'Tests blitting'
INFO:  01/08/24 18:14:01: Test Iteration 1: execKey 13487443248884952024
INFO:  01/08/24 18:14:01: Assert 'SDL_CreateWindow()': Passed
INFO:  01/08/24 18:14:01: Assert 'Check SDL_CreateWindow result': Passed
INFO:  01/08/24 18:14:01: Assert 'SDL_CreateRenderer()': Passed
ERROR: 01/08/24 18:14:01: Assert 'Check SDL_CreateRenderer result': Failed
ERROR: 01/08/24 18:14:01: >>> Suite Setup 'Render': Failed
INFO:  01/08/24 18:14:01: Total Test runtime: 0.0 sec
INFO:  01/08/24 18:14:01: ----- Test Case 14.5: 'render_testBlitColor' started
INFO:  01/08/24 18:14:01: Test Description: 'Tests blitting with color'
INFO:  01/08/24 18:14:01: Test Iteration 1: execKey 6545238456291081755
INFO:  01/08/24 18:14:01: Assert 'SDL_CreateWindow()': Passed
INFO:  01/08/24 18:14:01: Assert 'Check SDL_CreateWindow result': Passed
INFO:  01/08/24 18:14:01: Assert 'SDL_CreateRenderer()': Passed
ERROR: 01/08/24 18:14:01: Assert 'Check SDL_CreateRenderer result': Failed
ERROR: 01/08/24 18:14:01: >>> Suite Setup 'Render': Failed
INFO:  01/08/24 18:14:01: Total Test runtime: 0.0 sec
INFO:  01/08/24 18:14:01: ----- Test Case 14.6: 'render_testBlitAlpha' started
INFO:  01/08/24 18:14:01: Test Description: 'Tests blitting with alpha'
INFO:  01/08/24 18:14:01: Test Iteration 1: execKey 607625883004778424
INFO:  01/08/24 18:14:01: >>> Test 'render_testBlitAlpha': Skipped (Disabled)
INFO:  01/08/24 18:14:01: Total Test runtime: 0.0 sec
INFO:  01/08/24 18:14:01: ----- Test Case 14.7: 'render_testBlitBlend' started
INFO:  01/08/24 18:14:01: Test Description: 'Tests blitting with blending'
INFO:  01/08/24 18:14:01: Test Iteration 1: execKey 8246923526049529872
INFO:  01/08/24 18:14:01: >>> Test 'render_testBlitBlend': Skipped (Disabled)
INFO:  01/08/24 18:14:01: Total Test runtime: 0.0 sec
INFO:  01/08/24 18:14:01: Total Suite runtime: 0.0 sec
ERROR: 01/08/24 18:14:01: Suite Summary: Total=7 Passed=0 Failed=4 Skipped=3
ERROR: 01/08/24 18:14:01: >>> Suite 'Render': Failed
INFO:  01/08/24 18:14:01: ===== Test Suite 15: 'RWops' started
...
ERROR: 01/08/24 18:14:01: Run Summary: Total=305 Passed=289 Failed=4 Skipped=12
ERROR: 01/08/24 18:14:01: >>> Run /w seed 'TA2SFHIYGCCQXNDA': Failed
```

But for whatever reason, I don't get instructions at the end for re-running with `--filter render_whatever`?

**Comment by smcv** [comment]
> I'll make sure the test suite as a whole runs with dummy audio and video for 2.30 release

That would be great if you can, but it doesn't *necessarily* need to be a release blocker for "real SDL2": I don't think 100% of `testautomation` worked as acceptance tests in 2.28.x either (which is why it isn't included in Autotools `$(noninteractive)` or CMake `${SDL_TESTS_NONINTERACTIVE}` yet).

**Comment by smcv** [comment]
> But for whatever reason, I don't get instructions at the end for re-running with `--filter render_whatever`?

Oh, I see why - it's the shared setup function for the Render suite that is failing here, and not the individual tests within it.

**Comment by smcv** [comment]
Cherry-picking #8803 makes this a little clearer:

```
INFO:  01/08/24 18:36:26: Assert 'SDL_CreateRenderer()': Passed
ERROR: 01/08/24 18:36:26: Assert 'Check SDL_CreateRenderer result: Couldn't find matching render driver': Failed
```

**Comment by smcv** [comment]
SDL3 fixed this in 85e3099b. Backport available in #8804.

**Comment by smcv** [comment]
With #8804, I can get `testautomation` to pass by using `TEST_DISABLED` to stub out the 4 failing tests from https://github.com/libsdl-org/SDL/issues/8796, https://github.com/libsdl-org/SDL/issues/8797.
